internal/service/ec2/transit_gateway_multicast_domain_test.go

```diff
@@ -0,0 +1,1007 @@
+}
+resource "aws_ec2_transit_gateway_vpc_attachment" "test1" {
+  subnet_ids         = [aws_subnet.test1.id]
+  transit_gateway_id = aws_ec2_transit_gateway.test.id
+  vpc_id             = aws_vpc.test1.id
+  tags = {
+    Name = %[1]q
+  }
+}
+resource "aws_ec2_transit_gateway_multicast_domain" "test" {
+  transit_gateway_id = aws_ec2_transit_gateway.test.id
+
+  static_source_support = "enable"
+
+  association {
+    transit_gateway_attachment_id = aws_ec2_transit_gateway_vpc_attachment.test1.id
+    subnet_ids                    = [aws_subnet.test1.id]
+  }
+  members {
+    group_ip_address      = "224.0.0.1"
+    network_interface_ids = [aws_instance.test1.primary_network_interface_id, aws_instance.test2.primary_network_interface_id]
+  }
+  sources {
+    group_ip_address      = "224.0.0.1"
+    network_interface_ids = [aws_instance.test1.primary_network_interface_id]
+  }
+  tags = {
+    Name = %[1]q
+  }
+}
+`, rName)
+}
+
+func testAccTransitGatewayMulticastDomainConfigGroup3(rName string) string {
+	return fmt.Sprintf(`
+data "aws_availability_zones" "available" {
+  state = "available"
+}
+
+data "aws_ami" "amazon_linux" {
+  most_recent = true
+  owners      = ["amazon"]
+
+  filter {
+    name = "name"
+    values = [
+      "amzn-ami-hvm-*-x86_64-gp2",
+    ]
+  }
+
+  filter {
+    name = "owner-alias"
+    values = [
+      "amazon",
+    ]
+  }
+}
+resource "aws_vpc" "test1" {
+  cidr_block = "10.0.0.0/16"
+  tags = {
+    Name = %[1]q
+  }
+}
+resource "aws_subnet" "test1" {
+  vpc_id            = aws_vpc.test1.id
+  cidr_block        = "10.0.1.0/24"
+  availability_zone = data.aws_availability_zones.available.names[0]
+  tags = {
+    Name = %[1]q
+  }
+}
+resource "aws_instance" "test1" {
+  ami           = data.aws_ami.amazon_linux.id
+  instance_type = "t2.micro"
+  subnet_id     = aws_subnet.test1.id
+  tags = {
+    Name = %[1]q
+  }
+  lifecycle {
+    ignore_changes = [
+      iam_instance_profile,
+      tags,
+      tags_all,
+    ]
+  }
+}
+resource "aws_instance" "test2" {
+  ami           = data.aws_ami.amazon_linux.id
+  instance_type = "t2.micro"
+  subnet_id     = aws_subnet.test1.id
+  tags = {
+    Name = %[1]q
+  }
+  lifecycle {
+    ignore_changes = [
+      iam_instance_profile,
+      tags,
+      tags_all,
+    ]
+  }
+}
+resource "aws_ec2_transit_gateway" "test" {
+  multicast_support = "enable"
+  tags = {
+    Name = %[1]q
+  }
+}
+resource "aws_ec2_transit_gateway_vpc_attachment" "test1" {
+  subnet_ids         = [aws_subnet.test1.id]
+  transit_gateway_id = aws_ec2_transit_gateway.test.id
+  vpc_id             = aws_vpc.test1.id
+  tags = {
+    Name = %[1]q
+  }
+}
+resource "aws_ec2_transit_gateway_multicast_domain" "test" {
+  transit_gateway_id = aws_ec2_transit_gateway.test.id
+
+  static_source_support = "enable"
+
+  association {
+    transit_gateway_attachment_id = aws_ec2_transit_gateway_vpc_attachment.test1.id
+    subnet_ids                    = [aws_subnet.test1.id]
+  }
+  members {
```

Comment:
This type of multiple `members` as siblings feels unintuitive. I would rather see a members { type list }.

@ewbankkit thoughts?

The same would apply to `associations` and `sources`.